pkg/volume/csi/csi_attacher.go

```diff
@@ -650,15 +650,29 @@ func isAttachmentName(unknownString string) bool {
 
 func makeDeviceMountPath(plugin *csiPlugin, spec *volume.Spec) (string, error) {
 	if spec == nil {
-		return "", errors.New(log("makeDeviceMountPath failed, spec is nil"))
+		return "", errors.New("makeDeviceMountPath failed, spec is nil")
 	}
 
-	pvName := spec.PersistentVolume.Name
-	if pvName == "" {
-		return "", errors.New(log("makeDeviceMountPath failed, pv name empty"))
+	driver, err := GetCSIDriverName(spec)
+	if err != nil {
+		return "", err
+	}
+	if driver == "" {
+		return "", errors.New("makeDeviceMountPath failed, csi source driver name is empty")
+	}
+
+	csiSource, err := getPVSourceFromSpec(spec)
+	if err != nil {
+		return "", errors.New(log("makeDeviceMountPath failed to get CSIPersistentVolumeSource: %v", err))
+	}
+
+	if csiSource.VolumeHandle == "" {
+		return "", errors.New("makeDeviceMountPath failed to get CSIPersistentVolumeSource volume handle")
```

Comment:
how about error message "VolumeHandle is empty"?

Reply:
Can you please elaborate the exact expected message?

Reply:
Done